src/DataService/CachingPhpDocExtractor.php

```diff
@@ -0,0 +1,245 @@
+<?php
+
+namespace Lullabot\Mpx\DataService;
+
+use phpDocumentor\Reflection\DocBlock;
+use phpDocumentor\Reflection\DocBlockFactory;
+use phpDocumentor\Reflection\DocBlockFactoryInterface;
+use Symfony\Component\PropertyInfo\Extractor\ReflectionExtractor;
+use Symfony\Component\PropertyInfo\PropertyDescriptionExtractorInterface;
+use Symfony\Component\PropertyInfo\PropertyTypeExtractorInterface;
+use Symfony\Component\PropertyInfo\Type;
+use Symfony\Component\PropertyInfo\Util\PhpDocTypeHelper;
+
+/**
+ * Extracts data using a PHPDoc parser, using a cached context factory.
+ *
+ * @see \Symfony\Component\PropertyInfo\Extractor\PhpDocExtractor
+ */
+class CachingPhpDocExtractor implements PropertyDescriptionExtractorInterface, PropertyTypeExtractorInterface
+{
+    const PROPERTY = 0;
+    const ACCESSOR = 1;
+    const MUTATOR = 2;
+
+    /**
+     * @var DocBlock[]
+     */
+    private $docBlocks = [];
+
+    private $docBlockFactory;
+    private $contextFactory;
+    private $phpDocTypeHelper;
+    private $mutatorPrefixes;
+    private $accessorPrefixes;
+    private $arrayMutatorPrefixes;
+
+    /**
+     * @param DocBlockFactoryInterface $docBlockFactory
+     * @param string[]|null            $mutatorPrefixes
+     * @param string[]|null            $accessorPrefixes
+     * @param string[]|null            $arrayMutatorPrefixes
+     */
+    public function __construct(DocBlockFactoryInterface $docBlockFactory = null, array $mutatorPrefixes = null, array $accessorPrefixes = null, array $arrayMutatorPrefixes = null)
+    {
+        if (!class_exists(DocBlockFactory::class)) {
+            throw new \RuntimeException(sprintf('Unable to use the "%s" class as the "phpdocumentor/reflection-docblock" package is not installed.', __CLASS__));
+        }
+
+        $this->docBlockFactory = $docBlockFactory ?: DocBlockFactory::createInstance();
+        $this->contextFactory = new CachingContextFactory();
```

Comment:
This is why I had to copy this class. The upstream class declares this property as a private variable, which means we can't change it just by overriding the constructor. If only it was protected!